src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable, AreaCalculator {
+    private String name;
```

Comment:
Please remove this field, The only common field that figures should have is color regarding requirements, if you want to know the name of some figure you can use Object#getClass